src/database/queries/ApprovedItem.ts

```diff
@@ -85,6 +85,36 @@ export async function getApprovedItemByUrl(
   return db.approvedItem.findUnique({ where: { url } });
 }
 
+/**
+ *
+ *
+ * @param db
+ * @param externalId
+ * @param scheduledSurfaceGuid
+ * @param limit
+ */
+export async function getScheduledSurfaceHistory(
+  db: PrismaClient,
+  externalId: string,
+  scheduledSurfaceGuid?: string,
+  limit?: number
+): Promise<ScheduledItem[]> {
+  const approvedItem = db.approvedItem.findUnique({
+    where: { externalId },
+  });
+
+  return await db.scheduledItem.findMany({
+    take: limit,
+    orderBy: { scheduledDate: 'desc' },
+    where: {
+      approvedItemId: approvedItem.id,
```

Comment:
This line here breaks everything with a TS error at the moment. It's too late in the day to figure out what I'm doing wrong here.

Reply:
We just had to await the first db call when fetching the `approvedItem` and then do a non-null assertion, you were almost there!

Reply:
Can you not do joins with prisma?